### ISSUE TITLE
any-typed value in a TSX spread confuses --noUnusedLocals check

### ISSUE THREAD

**Issue by seansfkelley** [Microsoft/TypeScript#15779]
**TypeScript Version:**  2.3.2

**Code**

```tsx
import * as React from 'react';

export class Foo extends React.PureComponent<{}, {}> {
  render() {
    const foo = 'foo';
    return (
      <div
        {...({} as any)}
        title={foo}
      />
    );
  }
}
```

**Expected behavior:**

With `--noUnusedLocals`, the above code compiles without errors.

**Actual behavior:**

tsc complains:

```
5     const foo = 'foo';
            ~~~

Foo.tsx(5,11): error TS6133: 'foo' is declared but never used.
```

Changing the type of `Props#any` to anything else, such as `{ title: string }`, makes the errors go away.

**Comment by mhegazy** [comment]
Duplicate of https://github.com/Microsoft/TypeScript/issues/15478, should be fixed in TS 2.3.3.

The fix should be in `typescript@next` already, give it a try and let us know if you are still running into issues.

**Comment by seansfkelley** [comment]
Yup, no errors with `typescript@2.4.0-dev.20170511`. Thanks for the quick response!
